### ISSUE TITLE
Enable ToolResult Type in Core/AgentChat

### ISSUE THREAD

**Issue by victordibia** [microsoft/autogen#4725]
Currently, in Core/AgentChat the result of tool execution is mostly cast to a string ...
It would be good to standardize on some type e.g, 

- content
- attachments 
 - mime 
 - byte ..

Several benefits

- We can cast strucutred responses into messages that other agents/teams can "see" .e.g, inform them of artifacts created from tool call. 
- Interfaces like AGS can display such artifacts 

## How 

- Add toolresult type 
- Update core infra for tool call execution to check for structured output and honor them 
- Update things like AssistantAgent to parse structured output into inner_messages etc?

**Comment by ekzhu** [comment]
Yes, this should be added, similar to the structured message type #5131. Though I would consider instead of going for mime type we can stay in Python land for now. 

When we type hint a function tool, the return type can be a BaseModel. Thus it is possible to capture the result as part of the structured tool result message, with type information embedded in the message itself. 

@jackgerrits  some ideas on this? This will enable many scenarios.
